### PR TITLE
elinks: bump to newest version, thereby making it usable again

### PR DESCRIPTION
**Pull request by iblech**
###### Motivation for this change

elinks is a text-mode browser from the old times. I vividly remember that it was my main browser at some point. Alas, the current nixpkgs version of elinks is not really usable, as it fails with SSL errors on most sites. You can reproduce this yourself with

    $ nix-shell -p elinks
    $ elinks https://www.explainxkcd.com/wiki/index.php/Main_Page

or with most any site really.

This pull request packages the newest version, rendering it usable again. Since development of the official elinks repository seems to have paused, this pull request follows Debian in packaging felinks, a fork of elinks. We also adopt Debian's patches which make the build more reproducible and which implement defaults more adapted to today's age (for instance, to honor the caching behavior prescribed by web servers).

Since the nixpkgs package seemed to be unmaintained, I added myself as maintainer.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/networking/browsers/elinks/default.nix [hunk 1]
```diff
@@ -36,18 +36,24 @@ stdenv.mkDerivation {
...
     ++ stdenv.lib.optional enablePerl         "--with-perl"
     ++ stdenv.lib.optional enableSpidermonkey "--with-spidermonkey=${spidermonkey}"
     ;
+  
+  preConfigure = ''
+    patchShebangs ./autogen.sh
+    ./autogen.sh
+  '';
 
   meta = {
```
Comment (IvarWithoutBones) [inline pkgs/applications/networking/browsers/elinks/default.nix:52]: Currently ofborg fails to evaluate this, as `maintainers` isn't in scope.
```suggestion
  meta = with stdenv.lib; {
    description = "Full-featured text-mode web browser (package based on the fork felinks)";
    homepage = "https://github.com/rkd77/felinks";
    license = licenses.gpl2;
    platforms = platforms.linux;
    maintainers = with maintainers; [ iblech ];
  };
```

Comment (iblech) [reply pkgs/applications/networking/browsers/elinks/default.nix]: Oops, thank you, I added the maintainers field at the last moment and didn't notice that it wasn't in scope. I also tried `platforms.unix`, I cannot test on macOS, but Hydra can, let's see!

**Review by IvarWithoutBones — APPROVED** [review]

**Comment by Ma27** [comment]
Conflicts with master atm.

**Comment by iblech** [comment]
Thank you @Ma27, I just rebases to latest master and re-tested (alas only on Linux, not on macOS). I believe Borg will test darwin.

**Comment by xeji** [comment]
@GrahamcOfBorg build elinks

**Review by xeji — CHANGES_REQUESTED** [review]
elinks builds and works fine but it seems you accidentally included an unrelated commit 1a48e59 . Please remove it.

**Comment by iblech** [comment]
Oh, thanks for noticing! I just removed the stray commit. :-)

**Review by xeji — APPROVED** [review]

**Pull request merged**